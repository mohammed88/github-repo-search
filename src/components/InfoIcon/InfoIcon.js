import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Spacing from '../../ui/styles/spacing';
import {darkGray} from '../../ui/styles/colors';


const Wrapper = styled.div`
	align-items: center;
	display: flex;
	margin-right: ${Spacing.large};
`;

const IconText = styled.span`
	margin-left: 4px;
	color: ${darkGray};
`;

function InfoIcon({icon, text}) {
	return (
		<Wrapper>
			{icon}
			<IconText>{text}</IconText>
		</Wrapper>
	);
}

InfoIcon.propTypes = {
	icon: PropTypes.object.isRequired,
	text: PropTypes.any.isRequired,
};

export default InfoIcon;