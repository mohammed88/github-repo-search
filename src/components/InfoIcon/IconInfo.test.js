import React from 'react';
import {shallow} from 'enzyme';

import InfoIcon from './InfoIcon';

describe('InfoIcon', () => {
	it('should render', () => {
		const wrapper = shallow(<InfoIcon icon={{}} text='' />);

		expect(wrapper).toMatchSnapshot();
	});
});