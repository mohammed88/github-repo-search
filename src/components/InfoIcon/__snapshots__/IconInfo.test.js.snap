// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoIcon should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoIcon
    icon={Object {}}
    text=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        Object {},
        <ForwardRef>
          
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {},
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
	margin-left: 4px;
	color: ",
              "#4A4A4A",
              ";
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
	align-items: center;
	display: flex;
	margin-right: ",
          "24px",
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          Object {},
          <ForwardRef>
            
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {},
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
	margin-left: 4px;
	color: ",
                "#4A4A4A",
                ";
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
	align-items: center;
	display: flex;
	margin-right: ",
            "24px",
            ";
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
